### PR TITLE
Human chat: combined media captions, composer drop, redaction & scroll

### PR DESCRIPTION
**Pull request by webguru-hypha**
CI: CodeQL was failing on **ReDoS** (`trimUrlMatch` regex `/[),.;:]+$/u` in composer URL highlight). Replaced with **linear** trailing punctuation stripping.

Commit: `fix(human-chat): avoid ReDoS in composer URL punctuation trim` — CodeQL + check-types verified green on head.

<div><a href="https://cursor.com/agents/bc-57e4569a-6223-4f1e-8bd9-1b379accc153"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-57e4569a-6223-4f1e-8bd9-1b379accc153"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

## Summary by CodeRabbit

* **New Features**
  * Audio attachments, voice-message recording, dictation, mic dropdown, embeddable emoji picker surface, and drag‑and‑drop with visual drop prompt

* **Improvements**
  * Edit media messages while preserving existing media slots and adding/replacing attachments
  * Rich-media captions/replies, improved video playback UI and audio timeline/player
  * Redaction-aware timeline with immediate removal of deleted content and smarter auto-scroll
  * Reply headers now show author avatar previews

* **Localization**
  * Added composer, dictation/voice, and video control strings in en/de/es/fr/pt

### CODE REVIEW

**Review by alexprate — APPROVED** [review]

**Review by alexprate — APPROVED** [review]

File: packages/core/src/matrix/client/providers/matrix-provider.tsx [hunk 1]
```diff
@@ -677,15 +774,247 @@ export const MatrixProvider: React.FC<MatrixProviderProps> = ({ children }) => {
...
+          } as HyphaMediaEventContent;
+        } else {
+          const fn =
+            slots[0]?.filename?.trim() ||
+            String(rootFromSlot.body ?? 'attachment');
+          combined = {
+            ...combined,
+            body: fn,
+          } as HyphaMediaEventContent;
+        }
```
[the remark this answers is not in the document]
Comment (alexprate) [reply packages/core/src/matrix/client/providers/matrix-provider.tsx]: CodeRabbit threads (PR #2157)
There was one unresolved thread from coderabbitai: add tests for editing a media message with an empty caption while keeping the reply, covering the path that uses buildRichReplyMatrixContent(..., ' ') and keeps m.relates_to → m.in_reply_to.

What I did
That behavior is already covered by applyMediaEditCaptionAndReply (the same helper matrix-provider uses). I tightened the tests so they match CodeRabbit’s intent:

edit-room-message-media-caption.test.ts — After the empty-caption + reply case, assert splitRichReplyPlainBody(out.body).reply === '' (space placeholder trims to no visible reply) and keep the m.in_reply_to check.
chat-markup.test.ts — For buildRichReplyMatrixContent(..., ' '), import splitRichReplyPlainBody and assert the split the same way.
Tests run: pnpm test in @hypha-platform/core for those files — all passed.

Git
Commit: b00fbfe60 — test(core): tighten media edit empty-caption reply regression tests
Branch: pushed to cursor/human-chat-caption-dnd-scroll-c153
PR #2157 updated (via the integration)
Posting a PR comment from this environment failed with Resource not accessible by integration, so there is no automated summary comment on the PR; you can paste the summary above if you want it visible there.
